Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py

```diff
@@ -0,0 +1,1813 @@
+        table_headers = ['id', 'description', 'thumbprint', 'count', 'firstSeenAt', 'lastSeenAt']
+        object_data = {}
+        if result:
+            object_data = create_detected_exploit_output(result)
+
+        readable_output = tableToMarkdown(name='Found Detected Exploit:', t=object_data,
+                                          headers=table_headers, removeNull=True)
+        return CommandResults(outputs_key_field='id',
+                              outputs_prefix='SophosCentral.DetectedExploit',
+                              raw_response=result, outputs=object_data,
+                              readable_output=readable_output)
+    except DemistoException:
+        return CommandResults(f'Unable to find exploit: {detected_exploit_id}')
+
+
+def fetch_incidents(client: Client, last_run: Dict[str, int],
+                    first_fetch_time: str, fetch_severity: Optional[List[str]],
+                    fetch_category: Optional[List[str]],
+                    max_fetch: Optional[int]) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    Fetch incidents (alerts) each minute (by default).
+    Args:
+        client (Client): Sophos Central Client.
+        last_run (dict): Dict with last_fetch object,
+                                  saving the last fetch time(in millisecond timestamp).
+        first_fetch_time (dict): Dict with first fetch time in str (ex: 3 days ago).
+        fetch_severity (list(str)): Severity to fetch.
+        fetch_category (list(str)): Category(s) to fetch.
+        max_fetch (int): Max number of alerts to fetch.
+    Returns:
+        Tuple of next_run (millisecond timestamp) and the incidents list
+    """
+    last_fetch_timestamp = last_run.get('last_fetch', None)
+
+    if last_fetch_timestamp:
+        last_fetch_date = datetime.fromtimestamp(last_fetch_timestamp / 1000)
+        last_fetch = last_fetch_date
+    else:
+        first_fetch_date = dateparser.parse(first_fetch_time).replace(tzinfo=None)
+        last_fetch = first_fetch_date
+    incidents = []
+    next_run = last_fetch
+    alerts = client.search_alert(last_fetch.strftime(DATE_FORMAT)[:-4] + 'Z', None, None,
+                                 fetch_category, None, fetch_severity, None, max_fetch)
+    data_fields = ['id', 'description', 'severity', 'raisedAt', 'allowedActions', 'managedAgentId',
+                   'category', 'type']
+    for alert in alerts.get('items', []):
+        alert = create_alert_output(alert, data_fields)
+        alert_created_time = alert.get('raisedAt')
+        alert_id = alert.get('id')
+        incident = {
+            'name': f'Sophos Central Alert {alert_id}',
+            'occurred': alert_created_time,
+            'rawJSON': json.dumps(alert)
+        }
+        incidents.append(incident)
+        created_time = datetime.strptime(alert_created_time, DATE_FORMAT)
+        if created_time > next_run:
+            next_run = created_time

```

Comment:
is the  `alert_created_time` in timestamp?
doing `strptime` on every alert takes time, let's just rm that and `>` on the timestamp itself.
the `last_fetch ` should be of the last `created_time`.